src/lammpsio/dump.py

```diff
@@ -16,30 +16,89 @@
-    ``charge``, and ``mass``. The keys requiring 3 column indexes are ``position``,
-    ``velocity``, and ``image``. If a ``schema`` is not specified, it will be deduced
-    from the ``ITEM: ATOMS`` header.
-
-    The vector-valued fields (``position``, ``velocity``, ``image``) must contain all
-    three elements.
+    The LAMMPS dump file is a highly flexible file format. The ``schema`` of the
+    file is deduced from the ``ITEM: ATOMS`` header unless one is manually
+    specified. If specificed, the ``schema`` must be a dictionary mapping pieces
+    of data to column indexes (0-indexed). Valid keys for the schema match the
+    names and shapes in the `Snapshot`. The keys requiring only 1 column index
+    are:
+
+    - ``"id"``
+    - ``"typeid"``
+    - ``"molecule"``
+    - ``"charge"``
+    - ``"mass"``
+
+    The keys requiring 3 column indexes are:
+
+    - ``"position"``
+    - ``"velocity"``
+    - ``"image"``
+
+    LAMMPS dumps particles in an unknown order unless you have used the
+    ``dump_modify sort`` option. If you want particles to be ordered by ``id``
+    in the `Snapshot`, use ``sort_ids=True`` (default). Note that slightly
+    faster reading may be possible by setting this option to ``False``.
+
+    A `DumpFile` is iterable, so you can use it to go through all the snapshots
+    of a trajectory. Random access to snapshots is not currently implemented,
+    but it may be added in future. If you want to randomly access snapshots, you
+    should load the whole file into a list, but the memory requirements to do so
+    may be large.
+
+    The dump file may not contain certain information about your particles, for
+    example, topology, or you may choose not to write this information in the
+    dump file because it does not change frame-to-frame. The `copy_from` option
+    allows this information to be copied into a new snapshot from a reference
+    one, e.g, that was read from a `DataFile`.
 
     Parameters
     ----------
     filename : str
         Path to dump file.
     schema : dict
-        Schema for the contents of the file. Defaults to ``None``, which means to read
-        it from the file.
+        Schema for the contents of the file. Defaults to ``None``, which means
+        to read it from the file.
     sort_ids : bool
-        If true, sort the particles by ID in each snapshot.
-    copy_from : :class:`Snapshot`
-        If specified, copy fields that are missing in the dump file but are set in
-        a reference :class:`Snapshot`. The fields that can be copied are ``typeid``,
-        ``molecule``, ``charge``, and ``mass``.
+        If ``True``, sort the particles by ID in each snapshot.
+    copy_from : `Snapshot`
+        If specified, copy supported fields that are missing in the dump file
+        but are set in a reference `Snapshot`.
+
+    Example
+    -------
+    Create a dump file object:
+
+    .. code-block:: python
+
+        traj = lammpsio.DumpFile("atoms.lammpstrj")
```

Comment:
Would it be possible to make a tiny dump file in a hidden code block (e.g., at the top of the module) so that these snippets can be run under sybil?